src/dg/dg.cpp

```diff
@@ -3330,6 +3330,839 @@ real2 DGBase<dim,real,MeshType>::discontinuity_sensor(
+                    soln_basis.oneD_vol_operator);
+            }
+
+            for (unsigned int istate = 0; istate < nstate; ++istate) {
+                for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                    neigh_cell_avg[istate] += quad_weights[iquad]
+                        * soln_at_q_neigh[istate][iquad];
+                }
+            }
+
+            if (face_no == 0) {
+                prev_cell_avg = neigh_cell_avg;
+            }
+            else if (face_no == 1) {
+                next_cell_avg = neigh_cell_avg;
+            }
+        }
+
+        std::vector<dealii::types::global_dof_index> current_dofs_indices;
+        // Current reference element related to this physical cell
+        const int i_fele = soln_cell->active_fe_index();
+        const int poly_degree = i_fele;
+
+        const dealii::FESystem<dim, dim>& current_fe_ref = this->fe_collection[poly_degree];
+        const unsigned int n_dofs_curr_cell = current_fe_ref.n_dofs_per_cell();
+        // Obtain the mapping from local dof indices to global dof indices
+        current_dofs_indices.resize(n_dofs_curr_cell);
+        soln_cell->get_dof_indices(current_dofs_indices);
+
+        //Extract the local solution dofs in the cell from the global solution dofs
+        std::array<std::vector<real>, nstate> soln_dofs;
+        const unsigned int n_shape_fns = n_dofs_curr_cell / nstate;
+
+        for (unsigned int istate = 0; istate < nstate; ++istate) {
+            //allocate soln_dofs
+            soln_dofs[istate].resize(n_shape_fns);
+        }
+
+        for (unsigned int idof = 0; idof < n_dofs_curr_cell; ++idof) {
+            const unsigned int istate = this->fe_collection[poly_degree].system_to_component_index(idof).first;
+            const unsigned int ishape = this->fe_collection[poly_degree].system_to_component_index(idof).second;
+            soln_dofs[istate][ishape] = this->solution[current_dofs_indices[idof]]; //
+        }
+
+
+        const unsigned int n_quad_pts = this->volume_quadrature_collection[poly_degree].size();
+        const std::vector<real>& quad_weights = this->volume_quadrature_collection[poly_degree].get_weights();
+
+        //interpolate solution dofs to quadrature pts.
+        //and apply integral for the soln avg
+        std::array<std::vector<real>, nstate> soln_at_q;
+
+        for (int istate = 0; istate < nstate; istate++) {
+            soln_at_q[istate].resize(n_quad_pts);
+            soln_basis.matrix_vector_mult_1D(soln_dofs[istate], soln_at_q[istate],
+                soln_basis.oneD_vol_operator);
+        }
+
+        for (unsigned int istate = 0; istate < nstate; ++istate) {
+            for (unsigned int iquad = 0; iquad < n_quad_pts; ++iquad) {
+                soln_cell_avg[istate] += quad_weights[iquad]
+                    * soln_at_q[istate][iquad];
+            }
+        }
+
+        std::array<real, nstate> soln_cell_0 = {};
```

Comment:
You don't need to initialize with {}.

Reply:
Went through and changed all arrays to be initialized to zero using a for loop instead of using `={};`